model/src/clients/testsys_client.rs

```diff
@@ -0,0 +1,1301 @@
+                    })
+                {
+                    let next = next.to_string();
+                    channels = channels
+                        .into_iter()
+                        .map(|channel| replace_last(channel, next.clone()))
+                        .collect::<Vec<_>>();
+                    if let Err(e) = block_on(stream_tx.send(next.to_string())) {
+                        eprintln!("Some logs may have been lost: {}", e);
+                    }
+                    std::thread::sleep(Duration::from_millis(100));
+                }
+
+                if channels.iter().fold(true, |acc, (_, _, last, finished)| {
+                    acc && *finished && last.is_none()
+                }) {
+                    return;
+                }
+            }
+        });
+        Ok(stream_rx)
+    }
+
+    fn log_timestamp(log: &str) -> Option<NaiveDateTime> {
+        if let Ok(timestamp_regex) = Regex::new(r"^\[(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2})Z.*$") {
+            if let Some(time) = timestamp_regex
+                .captures(log)
+                .and_then(|captures| captures.get(1))
+            {
+                if let Ok(time) = NaiveDateTime::parse_from_str(time.as_str(), "%Y-%m-%dT%H:%M:%S")
+                {
+                    return Some(time);
+                }
+            }
+        }
+        None
+    }
+
+    async fn get_pods(&self, crd: &CrdName) -> Result<Vec<Pod>> {
+        let pod_api: Api<Pod> = self.namespaced_api();
+        Ok(match crd {
+            CrdName::Test(test) => {
+                pod_api
+                    .list(&ListParams {
+                        label_selector: Some(format!("job-name={}", test)),
+                        ..Default::default()
+                    })
+                    .await
+                    .context(KubeSnafu { action: "get pods" })?
+                    .items
+            }
+            CrdName::Resource(resource) => {
+                let mut pods = Vec::new();
+                pods.append(
+                    &mut pod_api
+                        .list(&ListParams {
+                            label_selector: Some(format!("job-name={}-creation", resource)),
+                            ..Default::default()
+                        })
+                        .await
+                        .context(KubeSnafu { action: "get pods" })?
+                        .items,
+                );
+                pods.append(
+                    &mut pod_api
+                        .list(&ListParams {
+                            label_selector: Some(format!("job-name={}-destruction", resource)),
+                            ..Default::default()
+                        })
+                        .await
+                        .context(KubeSnafu { action: "get pods" })?
+                        .items,
+                );
+                pods
+            }
+        })
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub enum CrdName {
```

Comment:
I'm looking for ways to reduce this Rust file's size. I think `Crd` and `CrdName` could go to its own file. Also it should be clearly documented what it is.